### ISSUE TITLE
Clarify philosophical background of Pledge

### ISSUE THREAD

**Issue by vladh** [opensourcepledge/opensourcepledge.com#227]
I think we should clarify the philosophical background of the Pledge more. There are two reasons for this.

1. I believe that having a much more solidly worked out philosophical position will help make our case to executives much stronger and more believable.

2. Questions come up in the community about our philosophical position. @voxpelli started [a thread on Mastodon][masto] with some questions that require an explanation of our economical position to be answered.

[masto]: https://fosstodon.org/@voxpelli@mastodon.social/113387125655068381

More specifically, what I'm talking about is a more detailed explanation of “Open Source is a restaurant”.

Financial markets mandate mutual disinterest. For the two of us to engage in a market interaction, you must not be concerned with my needs, and I must not be concerned with yours. If you want or need something of mine, I must withhold it (!) until you offer something of sufficient value in return. If I go to a hotel and explain that I would like them to allow me to stay in their room for free because I would really this, the hotel is within its rights to point out that, in the context of this market interaction, my desires are not relevant. I must offer something of sufficient value for the hotel to allow me to stay in their room. If the hotel just gives me the room for free in order to meet my needs, this is not a market interaction anymore. [^1]

On the other hand, Open Source software, and free software, treats the software as a gift. There is no need for me to withhold my creation until you offer something of sufficient value. I give you my work as a gift, along with the freedoms to use it as you please.

What we are saying is something like: we support the gift-giving in a sense — we do not ask for it to become a market interaction. However, at some point, payment must be made. Open Source is a restaurant — we don't mandate mutual disinterest, and the “food” is not withheld until you pay. But you must pay at some point.

Is this about care? Is this about an exchange between the person who writes the software and the person who pays? Is this about an exchange between the person who pays and the community more broadly?

We have to make this clearer in a longer-form post, building on @chadwhitacre's important work in Open Path.

[^1]: Waheed Hussain, “Living with the Invisible Hand: Markets, Corporations, and Human Freedom” (2023), sec A.2.5.

**Comment by voxpelli** [comment]
The “pay it forward” mentality plays into this, but some recent events has also shown some interesting new perspectives:

Matt Mullenweg believed WP Engine wasn’t contributing enough to WordPress / his company / something.

Matt Mullenweg is also very open with have become very rich from WordPress.

Without getting into that drama: When a maintainer can pay their rent, food and even a decent normal wage – does companies using their tool have a moral obligation to share even more with the maintainer? Or is the moral obligation rather to just ensure that the OSS projects one rely on are sustainable?

Important I think that the Pledge can not be perceived as a call to action to create more rich Matt Mullenwegs but instead a call to ensure a sustainable OSS ecosystem.

**Comment by vladh** [comment]
Yes, these are questions we must clearly answer. Thanks @voxpelli!

**Comment by chadwhitacre** [comment]
It's funny, I've been thinking about doing a follow-up post, "Open Source is _not_ a restaurant." In a restaurant you pay at the end, yes, but you order based on advertised prices. In Open Source there is none of that. The advertised price is free, so the expectation, the contract, is quite different.

That said, in my experience CEOs are generally swayed by things other than philosophical argument. 😬

**Comment by vladh** [comment]
> It's funny, I've been thinking about doing a follow-up post, "Open Source is _not_ a restaurant." In a restaurant you pay at the end, yes, but you order based on advertised prices. In Open Source there is none of that. The advertised price is free, so the expectation, the contract, is quite different.

Cool, I think this is important and I'd love to hear more about it/work it out together.

> That said, in my experience CEOs are generally swayed by things other than philosophical argument. 😬

For sure, I think we need to have a separate, simpler, more distilled line of arguments that we communicate to execs (and the general public!). But in order to do that, I think we ourselves need to have the more detailed version worked out, and the make the background reasoning available somewhere for when questions arise.

**Comment by vladh** [comment]
I'm adding to this to our priorities. I don't think we've done a good enough job at explaining the importance of _cash payments specifically_, as opposed to other kinds of contributions (ie [cash contributions allow maintainers to pay rent][phil] etc). We also need to do more work on making [our arguments clear][arguments], because this will help us in future conversations.

Examples of confusion arising from this:
* Recently: https://fosstodon.org/@voxpelli@mastodon.social/113414095977821484
* Previously: https://www.youtube.com/watch?v=QdUfB-hBoqk

[phil]: https://vlad.website/the-philosophy-of-the-open-source-pledge/
[arguments]: https://kinopio.club/arguments-nMca8dVCTI0pH4EZ2YXpY

**Comment by vladh** [comment]
Some excerpts from [things Alan Kay has said in the past][kay] are useful:

> Yet another perspective is to note that one of the human genetic "built-ins" is "hunting and gathering" -- this requires resources to "be around", and is essentially incremental in nature. It is not too much of an exaggeration to point out that most businesses are very like hunting-and-gathering processes, and think of their surrounds as resources put there by god or nature for them. Most don't think of the resources in our centuries as actually part of a human-made garden via inventions and cooperation, and that the garden has to be maintained and renewed.
> --
> As I pointed out in a previous email, Engelbart couldn't get funding from the very people who made fortunes from his inventions.
>
> It strikes me that many of the tech billionaires have already gotten their "upside" many times over from people like Engelbart and other researchers who were supported by ARPA, Parc, ONR, etc. Why would they insist on more upside, and that their money should be an "investment"? That isn't how the great inventions and fundamental technologies were created that eventually gave rise to the wealth that they tapped into after the fact.
>
> It would be really worth the while of people who do want to make money -- they think in terms of millions and billions -- to understand how the trillions -- those 3 and 4 extra zeros came about that they have tapped into. And to support that process.
> --
> The Parc money came from Xerox, the ARPA money came from DoD via the Cold War, but was unfettered and in the public domain. The most important difference between the "Golden Age" funders and those of today, is that the former didn't confuse responsibility with control -- they were responsible but they knew that the researchers had to control the choice of projects and methods. The funders of today -- most particularly the tech billionaires, but also execs in companies, bureaucrats in DARPA and NSF, etc -- think that they have to control. This winds up with bad choices for goals and projects, and bad processes. The "Golden Age" funders "funded people, not projects".
>
> If "good funding" were to come from the pop culture, I wouldn't turn up my nose at it (nor would any of my past colleagues). The main aim is to make qualitative improvements in the human condition. I think everyone would agree that making a billion dollars does not qualify a person to play professional sports, nor to be a classical violinist. Nor does it qualify a person to be able to direct fundamental research. These are all deep skills that anyone with a billion can learn, but if they don't learn them, then having them deep in the loop is a real problem for progress.

[kay]: https://worrydream.com/2017-12-30-alan/

**Comment by vladh** [comment]
I think some historical evidence can strengthen our arguments quite a lot. It's worth putting more effort into researching this.

**Comment by vladh** [comment]
Helpful resources in the economics literature:

* Open Source software is a [positive production externality][ppe]
* “[The Economics of Technology Sharing: Open Source and Beyond][tirole]” by Josh Lerner and Jean Tirole, particularly p 102–105
* “[The Success of Open Source][weber]” by Steven Weber, particularly p 149–156 (gift economy, public good, positive network externalities); p 227–231 (property rights); p 231–243 (distributed innovation); p 243–248 (tragedy of the commons, positive externalities, meaningful work)
* Open Source is not a gift economy (see Weber p 149–156)

[ppe]: https://en.wikipedia.org/wiki/Externality#Positive
[tirole]: https://www.aeaweb.org/articles?id=10.1257%2F0895330054048678
[weber]: https://www.hup.harvard.edu/books/9780674018587

**Comment by chadwhitacre** [comment]
Thanks for the references! Helpful for my upcoming talk at #283. I've ordered the two books and downloaded the Tirole paper. 👍

**Comment by vladh** [comment]
I've scheduled meetings with two economists to talk about this. I'll post updates here.

**Comment by chadwhitacre** [comment]
lol wow okay 😂

**Comment by vladh** [comment]
I've created a more comprehensive reading list — [Reading List: Economics of Public Goods][rl]

[rl]: https://github.com/opensourcepledge/docs/blob/main/reading-list-economics-of-public-goods.md

cc @chadwhitacre
